src/shared/dissect-image.c

```diff
@@ -1254,10 +1255,30 @@ int dissect_image_file(
 #endif
 }
 
+static bool dissected_image_has_mounted_btrfs_partition(DissectedImage *m) {
+        for (PartitionDesignator i = 0; i < _PARTITION_DESIGNATOR_MAX; i++) {
+                DissectedPartition *p = m->partitions + i;
+
+                if (p->mount_node_fd < 0)
+                        continue;
+
+                if (!streq_ptr(p->fstype, "btrfs") && !streq_ptr(p->decrypted_fstype, "btrfs"))
+                        continue;
+
+                return true;
+        }
+
+        return false;
+}
```

Comment:
why bother with this check btw? i mean, it's our device, so we won#t interfere with other people's stuff if we just call the ioctl always, and ignore if btrfs wasn't aware of the device anyway.